packages/containers/src/SidePanel/SidePanel.connect.js

```diff
@@ -1,39 +1,112 @@
 import get from 'lodash/get';
 import cmf, { cmfConnect } from '@talend/react-cmf';
-import { withRouter } from 'react-router';
 import Container, { DEFAULT_STATE } from './SidePanel.container';
+import { ACTION_TYPE_LINK } from './constants';
+
+const cache = {};
+
+/**
+ * Memoize per component on a key the result of the fn call
+ * @param {function} fn
+ * @param {string} componentId,
+ */
+function memoizeOne(fn, componentId = 'default', actions, currentRoute) {
+	if (!cache[componentId]) {
+		cache[componentId] = {};
+	}
+	const currentCache = cache[componentId];
+	if (currentCache.actions !== actions || currentCache.route !== currentRoute) {
+		currentCache.actions = actions;
+		currentCache.route = currentRoute;
+		currentCache.value = fn.apply();
+	}
+	return currentCache.value;
+}
+
+function isBasePathOf(pathToCheck, fullPath) {
+	const length = pathToCheck.length;
+	if (fullPath.length + 1 === length) {
+		return `${fullPath}/` === pathToCheck;
+	}
+	if (fullPath.length === length) {
+		return fullPath === pathToCheck;
+	}
+	if (fullPath.length < length) {
+		return false;
+	}
+	return pathToCheck === fullPath.slice(0, length) && fullPath[length] === '/';
+}
+
+function getActionsWrapped(actions) {
+	return actions.map(action => {
+		if (action.href && !action.onClick && !action.onClickDispatch && !action.onClickActionCreator) {
+			return {
+				...action,
+				onClick: event => {
+					event.preventDefault();
+					event.stopPropagation();
+				},
+				onClickDispatch: {
+					type: ACTION_TYPE_LINK,
+					cmf: {
+						routerPush: action.href,
+					},
+				},
+			};
+		}
+		return action;
+	});
+}
+
+function getSelectedAction(currentRoute, actions) {
+	return actions.find(action => action.href && isBasePathOf(action.href, currentRoute));
+}
+
+function getAction(id, currentRoute, state) {
+	const action = { actionId: id };
+	const info = cmf.action.getActionInfo(
+		{
+			registry: cmf.registry.getRegistry(),
+			store: {
+				getState: () => state,
+			},
+		},
+		id,
+	);
+	action.label = info.label;
+	action.id = info.id;
+
+	const route = get(
+		info,
+		'payload.cmf.routerReplace',
+		get(info, 'payload.cmf.routerPush', get(info, 'href')),
```

Comment:
Yes, but at the moment this task means extract the router first.
Then we will be able to work on containers for that.
This will not change the href implementation, we will just be able to get ride of that later